lib/index.d.ts

```diff
@@ -0,0 +1,528 @@
+    revocation_endpoint_auth_signing_alg?: string
+    tls_client_certificate_bound_access_tokens?: boolean
+
+    [key: string]: any
+  }
+
+  export interface IAuthorizationUrlParams {
+    redirect_uri?: string
+    response_type?: string
+    scope?: string,
+    [key: string]: any
+  }
+
+  export interface IEndSessionUrlParams {
+    id_token_hint?: TokenType
+    post_logout_redirect_uri?: string
+    state?: string
+  }
+
+  export type CallbackParamsType = {
+    state?: string
+    response_type?: string
+    nonce?: string
+    code_verifier?: string
+    [key: string]: any
+  }
+
+  export type TokenType = string | TokenSet
+
+  export interface ICallbackChecks {
+    /**
+     * When provided the authorization response will be checked for presence of required parameters for a
+     * given response_type. Use of this check is recommended.
+     */
+    response_type?: string
+    /**
+     * When provided the authorization response's state parameter will be checked to be the this expected one.
+     * Use of this check is required if you sent a state parameter into an authorization request.
+     */
+    state?: string
+    /**
+     * PKCE code_verifier to be sent to the token endpoint during code exchange. Use of this check is required
+     * if you sent a code_challenge parameter into an authorization request.
+     */
+    code_verifier?: string
+  }
+
+  export interface IExtendedCallbackChecks extends ICallbackChecks {
+    /**
+     * When provided the authorization response's ID Token auth_time parameter will be checked to be conform to the
+     * max_age value. Use of this check is required if you sent a max_age parameter into an authorization request.
+     */
+    max_age?: number
+    /**
+     * When provided the authorization response's ID Token nonce parameter will be checked to be the this expected
+     * one. Use of this check is required if you sent a nonce parameter into an authorization request.
+     */
+    nonce?: string
+  }
+
+  export interface ICallbackExtras {
+    /**
+     * extra request body properties to be sent to the AS during code exchange.
+     */
+    exchangeBody?: object
+    /**
+     * extra client assertion payload parameters to be sent as part of a client JWT assertion. This is only used
+     * when the client's token_endpoint_auth_method is either client_secret_jwt or private_key_jwt.
+     */
+    clientAssertionPayload?: object
+  }
+
+  export interface IRefreshExtras {
+    /**
+     * extra request body properties to be sent to the AS during refresh token exchange.
+     */
+    exchangeBody?: object
+    /**
+     * extra client assertion payload parameters to be sent as part of a client JWT assertion.
+     * This is only used when the client's token_endpoint_auth_method is either client_secret_jwt or private_key_jwt.
+     */
+    clientAssertionPayload?: object
+  }
+
+  export interface IGrantBody {
+    grant_type: string
+    [key: string]: any
```

Comment:
`unknown` is a safer type to use here, so the developer has to explicitly cast it to what they believe the type should be